.github/workflows/my-ci-pipeline.yml

```diff
@@ -7,12 +7,28 @@ on:
     paths-ignore:
       - 'READ.md'
 
-
 jobs:
   build:
     name: Build and test
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v2
     - run: docker build --tag todo-app:test --target test .
-    - run: docker run todo-app:test
\ No newline at end of file
+    - run: docker run todo-app:test
+
+  deploy:
+    name: Push to Docker Hub and Trigger App Service Deploy
+    runs-on: ubuntu-latest
+    needs: build
+    if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/exercise=9' }}
+    steps:
+    - uses: actions/checkout@v2
+    - run: echo hello deploy
+    - name: Login to Docker Hub
+      uses: docker/login-action@v3
+      with:
+        username: ${{ secrets.DOCKER_HUB_USERNAME }}
+        password: ${{ secrets.DOCKER_HUB_PASSWORD }}
+    - run: docker build --tag sidrahj/todo-app:latest --target production .
+    - run: docker push sidrahj/todo-app:latest
```

Comment:
Here you push `sidrahj/todo-app:latest` but your app service is actually using a different image named `sidrahj/hello-world:latest` which is why your latest change hasn't actually deployed to Azure! Given this name seems a better fit, my suggestion would be to change the image name in Azure (under the app -> Deployment Center) to `sidrahj/todo-app:latest`, and then try re-running your pipeline (which you can trigger manually, or by pushing another small text change to the index.html page)